iOS/ViewController.m

```diff
@@ -346,6 +352,9 @@ - (void)setARMode:(ARMode)mode {
     self.arMode = mode;
 
     if(wasEnabled != self.arEnabled) {
+        if(self.arEnabled) {
+            [self checkCameraAuthorization];
```

Comment:
It might be better to rely on the `ARSessionDelegate.session(_:didFailWithError:)` to detect when permissions and other things go wrong.

https://developer.apple.com/documentation/arkit/arsessionobserver/2887453-session
https://developer.apple.com/documentation/arkit/arerror